ntc_rosetta/parsers/openconfig/junos/__init__.py

```diff
@@ -18,9 +19,15 @@ class JunosParser(parser.RootParser):
 
     class Yangify(parser.ParserData):
         def init(self) -> None:
-            self.root_native["dev_conf"] = etree.fromstring(
-                self.root_native["dev_conf"]
-            )
+            parsed_xml = etree.fromstring(self.root_native["dev_conf"])
+            if parsed_xml.tag != "configuration":
+                parsed_xml = parsed_xml.find('configuration')
+                if parsed_xml is None: 
+                    raise AttributeError("Unable to locate 'configuration' tag in XML blob")
+                # We need to copy the XML element here, otherwise it will retain its parent 
```

Comment:
it’s not clear to me why this is needed. It looks like a NOOP. What potential problem are you trying to avoid?

Reply:
If no "configuration" tag is detected in the XML, then the configuration gets parsed, but an "empty" configuration is returned - as if the device configuration is empty. (this is what currently happens if you pass in a config that is wrapped in "rpc-reply" tags in the first comment)
I figured instead of failing silently, we should explicitly  notify the user that an invalid configuration was passed.